src/modules/azure.rs

```diff
@@ -0,0 +1,301 @@
+use encoding_rs::*;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read};
+use std::path::PathBuf;
+
+use super::{Context, Module, RootModuleConfig};
+
+type JValue = serde_json::Value;
+
+use crate::configs::azure::AzureConfig;
+use crate::formatter::StringFormatter;
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("azure");
+    let config = AzureConfig::try_load(module.config);
+
+    let subscription_name = get_azure_subscription_name(context)?;
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|variable, _| match variable {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "subscription" => Some(Ok(subscription_name.to_string())),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `azure`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_azure_config_file_location(context: &Context) -> Option<PathBuf> {
+    context
+        .get_env("AZURE_CONFIG_DIR")
+        .map(PathBuf::from)
+        .or_else(|| {
+            let mut home = dirs_next::home_dir()?;
```

Comment:
Please use `context.get_home()` and remove `dirs-next` from `Cargo.toml`
```suggestion
            let mut home = context.get_home()?;
```